src/ln/channelmanager.rs

```diff
@@ -1439,6 +1439,31 @@ impl ChannelManager {
 		Ok(accept_msg)
 	}
 
+	fn internal_accept_channel(&self, their_node_id: &PublicKey, msg: &msgs::AcceptChannel) -> Result<(), MsgHandleErrInternal> {
+		let (value, output_script, user_id) = {
+			let mut channel_state = self.channel_state.lock().unwrap();
+			match channel_state.by_id.get_mut(&msg.temporary_channel_id) {
+				Some(chan) => {
+					if chan.get_their_node_id() != *their_node_id {
+						//TODO: see issue #153 , need a consistent behavior on obnoxious behavior from random node
+						return Err(MsgHandleErrInternal::from_no_close(HandleError{err: "Got a message for a channel from the wrong node!", action: Some(msgs::ErrorAction::IgnoreError)}));
+					}
+					chan.accept_channel(&msg).map_err(|e| MsgHandleErrInternal::from_maybe_close(e))?;
+					(chan.get_value_satoshis(), chan.get_funding_redeemscript().to_v0_p2wsh(), chan.get_user_id())
+				},
+				None => return Err(MsgHandleErrInternal::from_no_close(HandleError{err: "Failed to find corresponding channel", action: Some(msgs::ErrorAction::IgnoreError)}))
```

Comment:
This should also be the ErrorMessage/Disconnect you had earlier, same as above (and also covered in #153).